application/src/PacificaSearchBundle/Controller/RestController.php

```diff
@@ -76,48 +47,43 @@ public function postFilterPageAction($type, $pageNumber, Request $request) : Res
     }
 
     /**
-     * Retrieves a page for each filter type based on the current contents of the filter
+     * Retrieves a page for each searchable type based on the contents of the filter
+     * @throws \Exception
      * @param Request $request
      * @return Response
      */
     public function postFilterPagesAction(Request $request) : Response
     {
         $filter = Filter::fromRequest($request);
+        $transactionIdsByFilterItem = $this->transactionRepository->getIdsByFilter($filter);
 
         $filterPages = [];
         foreach ($this->getFilterableRepositories() as $type => $repository) {
-            $filterPages[$type] = $repository->getFilteredPage(
-                $filter,
-                $this->getPageByType($type)
+            // Make a copy of the set of transactions but remove each type's own filter results - the contents of any
+            // given filter type are not constrained by what's already been chosen in that filter type, only by the
+            // selection of other filter types. For example, if you pick a user, you can still pick any other user, not
+            // only other users that share transactions with the original user.
+            $transIds = $transactionIdsByFilterItem;
+            unset($transIds[$type]);
```

Comment:
I think this must be a bug. Array assignment is by reference, so I'm unsetting $transactionIdsByFilterItem's value, too. Fix this.